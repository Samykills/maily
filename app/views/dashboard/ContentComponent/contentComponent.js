import React from "react";
import { View, Text, requireNativeComponent } from "react-native";
import { Card, CardItem, Icon, Footer } from "native-base";
import { AppColors, AppFonts } from "theme";
import { AppContext } from "react-native-easystore";
import { totalSize } from "react-native-dimension";
import ViewConstants from "../../constants/viewConstants";
const ContentView = requireNativeComponent("ContentView");
class ContentComponent extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { mail: "", isMarkedRead: "" };
  }

  contentComponentTrigger = (currentContext, appContext) => {
    this.setState({
      mail: appContext.selectedMail,
      isMarkedRead: appContext.selectedMail.isMarkedRead
    });
  };

  componentDidMount = () => {
    this.nativeUiListener = AppContext.initializeEventActivityListeners(
      this,
      this.contentComponentTrigger
    );
  };

  componentWillUnmount = () => {
    AppContext.removeEventActivityListeners(this.nativeUiListener);
  };

  _toggleMarkAsRead = value => {
    if (value) {
      return false;
    } else {
      return true;
    }
  };

  _onPressToggleReadButton = () => {
    let appContext = AppContext.getAppContext();
    appContext.selectedMail.isMarkedRead = this._toggleMarkAsRead(
      appContext.selectedMail.isMarkedRead
    );
    appContext.refreshEmailList = true;
    AppContext.setAppContext(appContext);
  };

  _renderContentHeader = () => {
    return (
      <CardItem header style={{ backgroundColor: AppColors.secondary }}>
        <Icon
          name="email"
          type="MaterialCommunityIcons"
          style={{ color: AppColors.selectionColor }}
        />
        <Text
          style={{
            color: AppColors.primaryFontColor,
            fontFamily: AppFonts.primaryFontFamily,
            fontSize: totalSize(1.8),
            fontWeight: "500"
          }}
        >
          {ViewConstants.LABELS.CONTENT_COMPONENT.HEADER}
        </Text>
      </CardItem>
    );
  };

  _renderContentBody = () => {
    let contentBody = this.state.mail ? (
      <ContentView
        style={{ flex: 1 }}
        buttonColor={AppColors.greenColor}
        buttonText={this.state.isMarkedRead ? "mark as Unread" : "mark as read"}
        text={this.state.mail.message}
        textSize={totalSize(2)}
        buttonTextColor={AppColors.whiteColor}
      />
    ) : (
      <CardItem style={{ justifyContent: "center" }}>
        <Text
          style={{
            color: AppColors.primaryFontColor,
            fontFamily: AppFonts.primaryFontFamily,
            fontSize: totalSize(1.8),
            fontWeight: "500"
          }}
        >
          {ViewConstants.LABELS.CONTENT_COMPONENT.INITIAL_MESSAGE}
        </Text>
      </CardItem>
    );
    return contentBody;
  };

  render() {
    return (
      <Card style={this.props.style}>
        {this._renderContentHeader()}
        {this._renderContentBody()}
      </Card>
    );
  }
}

export default ContentComponent;
