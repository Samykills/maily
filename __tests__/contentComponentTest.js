/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import "react-native";
import React from "react";
import ContentComponent from "../app/views/dashboard/contentComponent/contentComponent";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
jest.useFakeTimers();
it("contentView renders correctly", () => {
  const tree = renderer.create(<ContentComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
