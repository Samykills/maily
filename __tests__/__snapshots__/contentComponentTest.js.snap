// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentView renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    header={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        },
        Object {
          "backgroundColor": "#d1d8e0",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 28,
              "width": 35,
            },
            Object {
              "color": "#4b7bec",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="MaterialCommunityIcons"
    >
      ï‡®
    </Text>
    <Text
      style={
        Object {
          "color": "#000000BC",
          "fontFamily": undefined,
          "fontSize": 27.546799160701053,
          "fontWeight": "500",
        }
      }
    >
      Content
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000BC",
          "fontFamily": undefined,
          "fontSize": 27.546799160701053,
          "fontWeight": "500",
        }
      }
    >
      Click on mails above to view the content
    </Text>
  </View>
</View>
`;
